tests/test_init_and_backend.py

```diff
@@ -0,0 +1,139 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import filecmp
+import tempfile
+from pathlib import Path
+
+import mkosi
+import mkosi.backend as backend
+import pytest
+
+
+def test_workspace():
+    assert backend.workspace(Path("/home/folder/mkosi/mkosi")) == Path("/home/folder/mkosi")
+    assert backend.workspace(Path("/home/../home/folder/mkosi/mkosi")) == Path("/home/../home/folder/mkosi")
+    assert backend.workspace(Path("/")) == Path("/")
+    assert backend.workspace(Path()) == Path()
+
+def test_fedora_release_cmp():
+    assert mkosi.fedora_release_cmp("rawhide", "rawhide") == 0
+    assert mkosi.fedora_release_cmp("32", "32") == 0
+    assert mkosi.fedora_release_cmp("33", "32") > 0
+    assert mkosi.fedora_release_cmp("30", "31") < 0
+    assert mkosi.fedora_release_cmp("-1", "-2") > 0
+    assert mkosi.fedora_release_cmp("1", "-2") > 0
+    with pytest.raises(ValueError):
+        mkosi.fedora_release_cmp("literal", "rawhide")
+
+def test_strip_suffixes():
+    assert mkosi.strip_suffixes(Path("home/test.zstd")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home/test.xz")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home/test.raw")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home/test.tar")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home/test.cpio")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home/test.qcow2")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home.xz/test.xz")) == Path("home.xz/test")
+    assert mkosi.strip_suffixes(Path("home.xz/test")) == Path("home.xz/test")
+    assert mkosi.strip_suffixes(Path("home.xz/test.txt")) == Path("home.xz/test.txt")
+
+def test_copy_file():
+    with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
With pytest, we can simplify this by using the `tmpdir` fixture (see https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture). We don't need to create the temporary directory ourselves. Instead, pytest can do it for us.